### DISCUSSION TITLE
QFIELD CLOUD:  Data Pushed To Cloud Stuck at 'Pending Changes'

### DISCUSSION THREAD

**Discussion by JHL99** [opengisch/QField#2763] [Q&A]
**PROBLEM**
Changes pushed to cloud from devices are indicating as 'pending' but are not being added to the master data on the cloud.   Therefore, new data cannot be downloaded from the cloud. Using the commands on the web page to process the data are failing, per the log.

**BACKGROUND**
In late March/early April I started experimenting with QFIELD Cloud.  At that time, the cloud updates worked perfectly (both uploading and downloading to/from cloud), so I decided to invest a little more time in the QGIS project that I wanted to be cloud enabled.  In mid April, I enhanced the QGIS attributes for to use a pyQGIS function to assign a default value to a field.

I uploaded modified QGS project to the cloud, probably on April 15 or so.  The version was downloaded to an Andriod phone, which had previously worked flawlessly with QFIELD CLOUD.   I added some point data by using the Android phone and then pushed the changes to the cloud.  The cloud is reports 'Pending Changes'… no matter what I do (or what Nyuki Says) if I check the 'Jobs' status on the Cloud Web page, all of the 'Delta Apply' activities are 'Failed'.

Guessing that maybe the attribute form python code was the problem ( I don't know if it supported on QFIELD or not), I deleted it and uploaded my project again.  Testing revealed that  forcing 'pending changes' to update using the web page still failed.

Assuming something was wrong with QGS project, I finally created a new project using a geopackage with a single point layer (the previously working project also used gpkg) .  Uploaded it, put on mobile device, added a point and pushed changes to cloud, still  Pending changes failed. (see screen shots below)

Am I missing something?

Thanks for any guidance.
![image](https://user-images.githubusercontent.com/19516443/163689516-8425187d-3f3a-4a37-897c-d376a6c6ce7f.png)

![image](https://user-images.githubusercontent.com/19516443/163689529-ae13d312-caf6-4893-bb49-6ba314b7dfa3.png)

**Comment by suricactus** [comment]
Hey, @JHL99, thanks for asking. We are aware there is a problem and we are working on it. You can follow https://github.com/opengisch/qfieldcloud/issues/314 .

**Comment by JHL99** [comment]
Thanks for the reply and link, and eventual solution.

Jeff

________________________________
